means.py

```diff
@@ -22,10 +22,11 @@ def mean_rotor_in_chordal_metric(R, t=None):
     faster).
 
     """
-    if not t:
-        return np.quaternion(*(np.sum(as_float_array(R)))).normalized()
+    R_as_array = np.asarray(R, dtype=np.quaternion).view((np.double, 4))
+    if t is not None:
+        return np.quaternion(*sum(R_as_array)).normalized()
```

Comment:
Actually, this whole line should just be `return np.sum(R).normalized()`.  Here, `np.sum` will return a single quaternion.  More generally, you should avoid the plain python function `sum`, because if the input is an array of, say, a million quaternions, you've suddenly inserted a *python* loop with a million iterations, which is no good.